resolver/resolver.go

```diff
@@ -311,15 +287,9 @@ func (res *Resolver) HandleNonMesos(w dns.ResponseWriter, r *dns.Msg) {
 		// set refused
 		m.SetRcode(r, 5)
 	} else {
-
-		proto := "udp"
-		if _, ok := w.RemoteAddr().(*net.TCPAddr); ok {
```

Comment:
I'm don't think that using the same protocol as the incoming DNS message used to be transported is required when recursing. Please advise otherwise.